polyfill/lib/calendar.mjs

```diff
@@ -40,24 +39,21 @@ export class Calendar {
     if (ES.Type(fields) !== 'Object') throw new TypeError('invalid fields');
     options = ES.NormalizeOptionsObject(options);
     const overflow = ES.ToTemporalOverflow(options);
-    const { year, month, day } = impl[GetSlot(this, CALENDAR_ID)].dateFromFields(fields, overflow);
-    return new constructor(year, month, day, this);
+    return impl[GetSlot(this, CALENDAR_ID)].dateFromFields(fields, overflow, constructor, this);
   }
   yearMonthFromFields(fields, options, constructor) {
     if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
     if (ES.Type(fields) !== 'Object') throw new TypeError('invalid fields');
     options = ES.NormalizeOptionsObject(options);
     const overflow = ES.ToTemporalOverflow(options);
-    const { year, month } = impl[GetSlot(this, CALENDAR_ID)].yearMonthFromFields(fields, overflow);
-    return new constructor(year, month, this, /* referenceISODay = */ 1);
+    return impl[GetSlot(this, CALENDAR_ID)].yearMonthFromFields(fields, overflow, constructor, this);
```

Comment:
The other reason to push construction down is to enable non-ISO calendars to cache related data in a WeakMap. To do that the calendar implementation needs to know the constructed instance. I didn't want to do any caching in the top-level ISO implementation because it wasn't needed. So I'd prefer to leave as-is with the `impl` code being responsible for construction. What do you think?

An alternative could be adding something like `impl[calendar].cache(instance)` but that seems like encapsulation overkill. (Hmmm, that sounds like the name of the nerdiest punk band ever!)

Reply:
OK, got it, I understand why the calendar needs to have the constructed object now. Thanks!